SYCLStream.cpp

```diff
@@ -47,7 +47,15 @@ SYCLStream<T>::SYCLStream(const unsigned int ARRAY_SIZE, const int device_index)
   std::cout << "Driver: " << getDeviceDriver(device_index) << std::endl;
   std::cout << "Reduction kernel config: " << dot_num_groups << " groups of size " << dot_wgsize << std::endl;
 
-  queue = new cl::sycl::queue(dev);
+  queue = new cl::sycl::queue(dev, [&](cl::sycl::exception_list l) {
+      try {
+        for(auto e: l) {
+          std::rethrow_exception(e);
+        }
+      } catch (cl::sycl::exception e) {
+        std::cout << e.what();
```

Comment:
Would all exceptions be fatal? If so, I think we should exit/abort/rethrow here, otherwise the application will try and continue after printing the `sycl::exception`.